### PR TITLE
Add support for USE in cypher builder

### PR DESCRIPTION
**Pull request by angrykoala**
# Description
Support for `USE` clauses (https://neo4j.com/docs/cypher-manual/5/clauses/use/)

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 35 | 39 | 1134 |
|  aggregations.NestedAggregation | 14551 | 14551 | 101 | 122 | 2174 |
|  aggregations.AggregationWithWhere | 11979 | 11979 | 68 | 118 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 22116987 | 3457 | 3641 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116987 | 22116987 | 3088 | 3016 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9734 | 9734 | 70 | 71 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9937 | 9937 | 71 | 88 | 2174 |
|  aggregations.DeeplyNestedCount | 13070331 | 13070333 | 4665 | 5055 | 2008534 |
|  batch-create.BatchCreate | 4200 | 4200 | 199 | 167 | 600 |
|  connect.createAndConnect | 14424 | 14424 | 164 | 206 | 3003 |
|  connections.Connection | 14082 | 14082 | 77 | 81 | 2174 |
|  connections.NestedConnection | 41462 | 41462 | 114 | 171 | 4516 |
|  connections.ConnectionWithSort | 2282 | 2282 | 59 | 82 | 1040 |
|  connections.ConnectionWithSortAndCypher | 2282 | 2282 | 211 | 330 | 1040 |
|  create.SimpleMutation | 7 | 7 | 54 | 63 | 1 |
|  cypher-directive.TopLevelCypherDirective | 116092 | 116092 | 808 | 901 | 12241 |
|  cypher-directive.TopLevelCypherDirectiveWithColumnName | 147611 | 147611 | 305 | 418 | 12241 |
|  delete.SimpleDelete | 19401 | 19401 | 252 | 372 | 1040 |
|  delete.NestedDeleteInUpdate | 22200 | 22200 | 144 | 175 | 2040 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 40 | 66 | 1040 |
|  query.SimpleQuery | 15160 | 15160 | 88 | 102 | 2174 |
|  query.QueryWhere | 9699 | 9711 | 45 | 41 | 2162 |
|  query.SimpleQueryWithNestedWhere | 9877 | 9889 | 50 | 56 | 2162 |
|  query.Nested | 10092037 | 10092037 | 10768 | 11326 | 2008534 |
|  query.NestedWithFilter | 40401 | 40401 | 134 | 136 | 4511 |
|  query.OrFilterOnRelationships | 41302 | 42205 | 164 | 187 | 1252 |
|  query.OrFilterOnRelationshipsAndNested | 34961 | 35914 | 238 | 247 | 1252 |
|  query.QueryWithNestedIn | 13403 | 13349 | 58 | 77 | 1220 |
|  query.NestedConnectionWhere | 9834 | 9834 | 52 | 84 | 2174 |
|  query.DeeplyNestedConnectionWhere | 10029 | 10024 | 93 | 93 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 6343 | 6343 | 138 | 146 | 1134 |
|  query.NestedWithRelationshipSingleFilters | 3024400 | 3024400 | 611 | 639 | 1003942 |
|  query.Fulltext | 96 | 96 | 25 | 29 | 16 |
|  query.FulltextWithNestedQuery | 587 | 587 | 41 | 43 | 84 |
|  sorting.SortMultipleTypes | 2513 | 2513 | 79 | 155 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 1492 | 1469 | 166 | 367 | 1040 |
|  sorting.SortOnNestedFields | 14082 | 14082 | 105 | 112 | 2174 |
|  sorting.SortDeeplyNestedFields | 42196 | 42196 | 120 | 154 | 4516 |
|  sorting.SortWithTopLevelCypher | 2119 | 2119 | 174 | 191 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 115 | 80 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 65 | 68 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 330 | 360 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 314 | 345 | 33033 |
|  update.NestedUpdate | 2119 | 2119 | 65 | 68 | 1040 |

</details>
        
Old Schema Generation: 33.818s
Schema Generation: 34.104s
Old Subgraph Schema Generation: 49.698s
Subgraph Schema Generation: 50.400s

**Review by darrellwarde — APPROVED** [review]
I thought I reviewed this Friday! 🙈 LGTM!

File: .changeset/big-glasses-jam.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@neo4j/cypher-builder": patch
```
Comment (MacondoExpress) [inline .changeset/big-glasses-jam.md:2]: I would say that is a minor change!

**Review by MacondoExpress — APPROVED** [review]
LGTM!

File: .changeset/big-glasses-jam.md (hunk 1 above)
Comment (darrellwarde) [reply .changeset/big-glasses-jam.md:2]: But when a product is in `0.*.*`, doesn't a patch essentially mean minor, and a minor essentially mean major?

Comment (MacondoExpress) [reply .changeset/big-glasses-jam.md:2]: Absolutely, my bad!

**Pull request merged**